src/org/openstreetmap/josm/gui/dialogs/FilterDialog.java

```diff
@@ -175,10 +176,11 @@ private class DeleteAction extends FilterAction {
 
         @Override
         public void actionPerformed(ActionEvent e) {
-            int index = filterModel.getSelectionModel().getMinSelectionIndex();
-            if (index >= 0) {
-                filterModel.removeFilter(index);
-            }
+            // Remove in reverse order to preserve index validity
+            int[] selected = filterModel.getSelectedIndices();
+            Arrays.stream(selected).boxed()
+                .sorted(Collections.reverseOrder())
+                .forEachOrdered(filterModel::removeFilter);
```

Comment:
I really don't like this code.

FilterTableModel#fireTableRowsDeleted calls `savePrefs()`, `updateFilters()`, and `fireTableRowsDeleted` _every time_. This can get expensive quickly.

I'd much prefer to have a `removeFilters(int... rowIndexes)` method in FilterTableModel, which would look something like this:
```java
    /**
     * Removes the filters that is displayed in the given rows
     * @param rowIndex The index of the filters to remove
     * @since xxx
     */
    public void removeFilters(int... rowIndex) {
        boolean modified = false;
        Arrays.sort(rowIndex);
        for (int i = rowIndex.length - 1; i >= 0; i--) {
            modified |= model.removeFilter(i) != null;
        }
        if (modified) {
            savePrefs();
            updateFilters();
            List<int[]> groupedRows = groupRows(rowIndex);
            // Reverse to avoid having to deal with offsets
            Collections.reverse(groupedRows);
            for (int[] rows : groupedRows) {
                fireTableRowsDeleted(rows[0], rows[1]);
            }
        }
    }
```

Note: I added `groupRows` in https://josm.openstreetmap.de/changeset/18439/josm/, but it isn't available. It should probably be moved to Utils and renamed.